bitbots_vision/bitbots_vision/vision_modules/yoeo_handler.py

```diff
@@ -0,0 +1,207 @@
+from candidate import CandidateFinder, Candidate
+
+from abc import ABC, abstractmethod
+from collections import defaultdict
+import cv2
+from numpy import where, moveaxis
+from os.path import join
+from rclpy import logging
+from typing import List
+
+logger = logging.get_logger('vision_yoeo')
+
+try:
+    from yoeo import models as torch_models, detect as torch_detect
+    from yoeo.utils.utils import load_classes
+except ImportError:
+    logger.error("Not able to import pytorchyolo. This might be fine if you use another method.")
+
+
+class YOEOHandlerTemplate(ABC):
+    def __init__(self, config: dict, model_path: str):
+        self._use_caching = config['caching']
+
+        self._image = None
+        self._iou_non_max_suppression_thresh = config['yoeo_nms_threshold']
+
+        self._det_candidates = defaultdict(list)
+        self._det_class_names = None
+        self._det_confidence_thresh = config['yoeo_conf_threshold']
+
+        self._segmentation = None
+        self._segmentations = dict()
+        self._seg_class_idx = dict()
+
+        self._load_candidate_class_names(model_path)
+
+    def _load_candidate_class_names(self, model_path: str) -> None:
+        path = join(model_path, "data", "yoeo_names.yaml")
+        data = load_classes(path)
+
+        self._det_class_names = data['detection']
+        for key, i in zip(data['segmentation'], range(len(data['segmentation']))):
```

Comment:
Instead of a dict one could also use the position in the list as it should be eqivalent to the mapping above.

Reply:
> One could use
> 
> ```python
> for i, key in enumerate(data['segmentation']):
> ```
> 
> and maybe also a dict comprehension:
> 
> ```python
> self._seg_class_idx = {key: i for i, key in enumerate(data['segmentation'])}
> ```
> 
> or is there something i am missing?

Good point, I'll change that.

Reply:
> Instead of a dict one could also use the position in the list as it should be eqivalent to the mapping above.

Sure, but I think I list would just confuse. All we really need is the specified mapping. And at least theoretically, the complexity should be better with a dict (not that it would make that much of a difference with just 3 elements). I'd stick with a dict here if you don't mind?

Reply:
I agree that the complexity is most likely better with the dict, but for the object detector `self._det_class_names` is also only a list of names, which might be confusing, so I would like them to be the same data type. I know, that the mapping is accessed the other way around most of the time for the object detector, so it makes sense complexity wise, but it seems confusing nether the less.